### ISSUE TITLE
Metric depth recovery part

### ISSUE THREAD

**Issue by xyclone10** [YvanYin/Metric3D#153]
Hello! Thank you for your amazing work! I already read the paper and I might miss some information, I would appreciate your answers, but I have several questions:

1. For the backbone, as stated in the paper, you used DINO v2-reg as the encoder and DPT as the decoder. Did you freeze the encoder when training?
2. When preprocessing the data, did you normalize the dataset to relative depth before inputting it to the model? If so, which part of the pipeline emphasizes the recovery of metric depth? Which part from the source code did those roles?

As far as I know, DINOv2 normalized the input to extract the features. Thus, if we froze the encoder, the output from the DPT decoder would be in relative depth too. I am still new to this, so please correct me too if I am wrong :)

Thank you!

**Comment by JUGGHM** [comment]
> Hello! Thank you for your amazing work! I already read the paper and I might miss some information, I would appreciate your answers, thus I have several questions:
> 
> 1. For the backbone, as stated in the paper, you used DINO v2-reg as the encoder and DPT as the decoder. Did you freeze the encoder when training?
> 2. When preprocessing the data, did you normalize the dataset to relative depth before inputting it to the model? If so, which part of the pipeline emphasizes the recovery of metric depth? Which part from the source code did those roles?
> 3. As stated in the supplementary materials, you used depth bins ranging from [0.1m,200m] for ViT models. Could you elaborate more what kind of depth bins processing that you did?
> 
> As far as I know, DINOv2 normalized the input to extract the features. Thus, if we froze the encoder, the output from the DPT decoder would be in relative depth too. I am still new to this, so please correct me too if I am wrong :)
> 
> Thank you!

Thank you for your interest, I hope the following answers help:
(1) Not frozen. We observe the loss converges much faster and to a lower level if all the parameters are trained.
(2) No, but all metric depth maps are normalized according to the camera focal length ratio and size ratio. Please refer to class [LableScaleCanonical](https://github.com/YvanYin/Metric3D/blob/34afafe58d9543f13c01b65222255dab53333838/mono/utils/transform.py#L107) for more details.
(3) Uniform bins in log space, which is a common practice stressed in DORN (Fu etal CVPR2018). Please refer to its definition [here](https://github.com/YvanYin/Metric3D/blob/34afafe58d9543f13c01b65222255dab53333838/mono/model/decode_heads/RAFTDepthNormalDPTDecoder5.py#L794).
(4) A normalized feature input does not necessarily mean the predicted depth should also be normalized relative depth.

**Comment by MoAbbasid** [comment]
@JUGGHM Hi, Im trying to use your model for outdoor scene to get metric depth to eventually get a sense of scale at the specific location, but the depth values are too varied, can you help me with this [more details here](https://github.com/YvanYin/Metric3D/issues/161#issue-2535774407)

**Comment by xyclone10** [comment]
Hello!

I have some questions regarding the HDNL loss that you proposed.
So, I tried to train with AbsRel + SiLog + HDNL. But it didn't converge at all, does this loss have anything like special cases to be converged with other losses?

This is when the epoch is in training
![image](https://github.com/user-attachments/assets/4536649b-f02b-4765-89cc-efce2371266f)

After one epoch, it gave this kind of result
![image](https://github.com/user-attachments/assets/30d38dc2-76fd-47f3-aac6-27b724858fcb)

I already confirm that the AbsRel+SiLog won't affect this, I simply add 0.5 of HDNL and it didn't converge.

Thank you!

**Comment by YvanYin** [comment]
@xyclone10 Have you tried using a small data split for overfitting? When we train, it is easy to converge when we combine all losses.

**Comment by xyclone10** [comment]
> @xyclone10 Have you tried using a small data split for overfitting? When we train, it is easy to converge when we combine all losses.

I have tried just now and it is still same. For your information, my model output is depth regression. I wonder if it takes effect on it. Or is it a must to do uniform bins in log space to produce metric depth?
